src/pdf_parser/pdf_utils/ocr.py

```diff
@@ -1,11 +1,329 @@
 import concurrent.futures
+import io
+from collections import defaultdict
 from typing import Optional, Tuple, List, Union
 
 import numpy as np
-from layoutparser import TextBlock, Layout
+from PIL.PpmImagePlugin import PpmImageFile
+from google.cloud import vision
+from google.cloud.vision import types
+from google.cloud.vision_v1.types import BoundingPoly  # type: ignore
+from google.protobuf.pyext._message import RepeatedCompositeContainer
+from layoutparser import TextBlock, Rectangle, Layout  # type: ignore
 from layoutparser.ocr import TesseractAgent, GCVAgent
+from shapely.geometry import Polygon
 
 from src.base import PDFTextBlock
+from src.pdf_parser.pdf_utils.disambiguate_layout import lp_coords_to_shapely_polygon
+from src.pdf_parser.pdf_utils.utils import BaseModel
+
+
+# Hierarchical data structure for representing document text.
+class GoogleTextSegment(BaseModel):
+    """A segment of text from Google OCR."""
+
+    text: str
+    coordinates: BoundingPoly
+    confidence: float
+    language: Optional[str]
+
+
+class GoogleBlock(BaseModel):
+    """A fully structured block from google OCR. Can contain multiple segments."""
+
+    coordinates: BoundingPoly
+    text_blocks: List[GoogleTextSegment]
+
+
+def image_bytes(image: PpmImageFile) -> bytes:
+    """Return the image as a bytes object."""
+    image_buffer = io.BytesIO()
+    image.save(image_buffer, format="PNG")
+    image_buffer.seek(0)
+    return image_buffer.read()
+
+
+def google_vertex_to_shapely(bound):
+    return Polygon(
+        [
+            (bound.vertices[0].x, bound.vertices[0].y),
+            (bound.vertices[1].x, bound.vertices[1].y),
+            (bound.vertices[2].x, bound.vertices[2].y),
+            (bound.vertices[3].x, bound.vertices[3].y),
+        ]
+    )
+
+
+def extract_google_layout(
+    image: PpmImageFile,
+) -> Tuple[
+    List[GoogleBlock],
+    List[GoogleTextSegment],
+    List[GoogleTextSegment],
+    List[GoogleTextSegment],
+]:
+    """Returns document bounds given an image.
```

Comment:
Pretty sure this shouldn't pass a precommit check?

Reply:
It does. What's the issue you've spotted?

Reply:
Just assumed the args and returns indentation would fail but that's good if it doesn't!